### ISSUE TITLE
`finish` command doesn't work as expected inside autoloaded files

### ISSUE THREAD

**Issue by russelldavis** [deivid-rodriguez/byebug#290]
To reproduce:

``` ruby
# --- foo.rb ---
autoload :Bar, './bar'

def main
  Bar
  puts 'end of main'
end

main
puts 'end of foo.rb'
```

``` ruby
# --- bar.rb ---
require 'byebug'; byebug
module Bar
end
```

Run `ruby foo.rb`. When you hit the breakpoint, run `backtrace`. Note that byebug properly treats the `bar.rb` context as its own frame. Now, run `finish`. The expected/desired behavior is for byebug to break on the `'end of main'` line. Instead, it breaks on the `'end of foo.rb'` line.

I did some digging, and it looks like this is happening because the TracePoint API doesn't make any callbacks when autoloading begins or ends. Which means that `context->steps_out` never gets decremented when the autoload file finishes executing. When stepping into a `require` line, this isn't a problem, since we get a TracePoint callback for the `require` call and return.

Any thoughts on the best way to deal with this? Thanks.

**Comment by deivid-rodriguez** [comment]
Hi  @russelldavis! Sorry it took me so long to reply. Initial investigation shows this is probably a bug in the TracePoint API fixed in ruby 2.3! I'm adding a regression test for it in #405 and dropping official support for ruby 2.2 in there as well.

**Comment by deivid-rodriguez** [comment]
Closed by #405!
